lib/driver.js

```diff
@@ -182,78 +182,54 @@ class YouiEngineDriver extends BaseDriver {
     return true;
   }
 
-  async setupNewIOSDriver (opts) {
+  async setupNewIOSDriver (caps) {
     let iosArgs = {
       javascriptEnabled: true,
     };
 
     let iosdriver = new IOSDriver(iosArgs);
     // If iOS version is 10 or above we need to use XCUITestDriver (and Xcode 8+)
-    if (opts.platformVersion) {
-      let majorVer = opts.platformVersion.toString().split(".")[0];
+    if (caps.platformVersion) {
+      let majorVer = caps.platformVersion.toString().split(".")[0];
       if (parseInt(majorVer, 10) >= 10) {
         iosdriver = new XCUITestDriver(iosArgs);
       }
     }
-
-    let caps = {
-      platformName: "iOS",
-      browserName: "",
-      app: opts.app,
-      udid: opts.udid,
-      platformVersion: opts.platformVersion,
-      deviceName: opts.deviceName,
-      //XCUITest specific (no harm in sending them to IOSDriver)
-      noReset: opts.noReset,
-      processArguments: opts.processArguments,
-      wdaLocalPort: opts.wdaLocalPort,
-      showXcodeLog: opts.showXcodeLog,
-      realDeviceLogger: opts.realDeviceLogger,
-      iosInstallPause: opts.iosInstallPause,
-      xcodeConfigFile: opts.xcodeConfigFile,
-      // Disabling the proxy CommandTimeout in the iOS driver since we are now handling it in the YouiEngine Driver
-      newCommandTimeout: 0
-    };
+    // Disabling the proxy CommandTimeout in the iOS driver since we are now handling it in the YouiEngine Driver
+    caps.newCommandTimeout =  0
     await iosdriver.createSession(caps);
 
     return iosdriver;
   }
 
-  async startIOSSession () {
+  async startIOSSession (caps) {
     logger.info("Starting an IOS proxy session");
     this.proxyAllowList = TO_PROXY_IOS;
-    let opts = _.cloneDeep(this.opts);
+    let capsCopy = _.cloneDeep(caps);
```

Comment:
One reason is that we are modifying newCommandTimeout and without cloning it, we'd be modifying it for our driver as well.
It probably should be done inside setupNewIOSDriver instead.
Do you want it changed?